From 6d613d88acc146ef57fd310cc7ec72220613e7cf Mon Sep 17 00:00:00 2001
Message-Id: <patch-6.6-6d613d88acc-20210614T172422Z-avarab@gmail.com>
In-Reply-To: <cover-0.6-00000000000-20210614T172422Z-avarab@gmail.com>
References: <cover-0.6-00000000000-20210614T172422Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Thu, 22 Apr 2021 10:01:47 +0200
Subject: [PATCH 6/6] mktag tests: test fast-export
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Pass the bad tags we've created in the mktag tests through
fast-export, it will die on the bad object or ref, let's make sure
that happens.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 t/t3800-mktag.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/t/t3800-mktag.sh b/t/t3800-mktag.sh
index 7d0ad3c8e62..ad0da2cd2b2 100755
--- a/t/t3800-mktag.sh
+++ b/t/t3800-mktag.sh
@@ -85,6 +85,11 @@ check_verify_failure () {
 
 		test_must_fail git -C bad-tag for-each-ref --format="%(*objectname)"
 	'
+
+	test_expect_success "fast-export & fast-import: $subject" '
+		test_must_fail git -C bad-tag fast-export --all &&
+		test_must_fail git -C bad-tag fast-export $bad_tag
+	'
 }
 
 test_expect_mktag_success() {
-- 
2.32.0.555.g0268d380f7b

